build-tools/xa-prep-tasks/Xamarin.Android.BuildTools.PrepTasks/CheckoutExternalGitDependency.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.IO;
+using Microsoft.Build.Framework;
+
+namespace Xamarin.Android.BuildTools.PrepTasks
+{
+	public class CheckoutExternalGitDependency : Git
+	{
+		[Required]
+		public ITaskItem ExternalGitDependency { get; set; }
+
+		protected override bool LogTaskMessages {
+			get { return false; }
+		}
+
+		string commit;
+		string owner;
+		string name;
+
+		public override bool Execute ()
+		{
+			commit = ExternalGitDependency.ItemSpec;
+			owner = ExternalGitDependency.GetMetadata ("Owner");
+			name = ExternalGitDependency.GetMetadata ("Name");
+			string destination = Path.Combine (GetWorkingDirectory (), name);
+
+			if (!Directory.Exists (destination)) {
+				Clone ();
+			}
+
+			WorkingDirectory.ItemSpec = destination;
+			Fetch ();
+			CheckoutCommit ();
+
+			return !Log.HasLoggedErrors;
+		}
+
+		void Clone ()
+		{
+			string ghToken = Environment.GetEnvironmentVariable("GH_AUTH_SECRET");
+			if (!string.IsNullOrEmpty (ghToken)) {
+				Arguments = $"clone https://{ghToken}@github.com/{owner}/{name} --progress";
+			} else {
+				// Fallback to SSH URI
+				Arguments = $"clone git@github.com:{owner}/{name} --progress";
+			}
+
```

Comment:
a'la:

```csharp
Arguments += " " + Path.GetFullPath (destination);
```

(though `Clone()` would need to be updated to receive `destination` as a parameter.)